src/components/accounts/SetupImplicit.js

```diff
@@ -203,22 +216,35 @@ class SetupImplicit extends Component {
         });
     }
 
+    handleFundWithAccount = async () => {
+        const { dispatch, implicitAccountId, newAccountId, recoveryMethod } = this.props
+        this.setState({ createFromAccount: true })
+        await dispatch(sendMoney(implicitAccountId, MIN_BALANCE_TO_CREATE.toString()))
+        await dispatch(createAccountFromImplicit(newAccountId, implicitAccountId, recoveryMethod))
```

Comment:
I think we shouldn't send money to implicit account first

```
export const createAccountFromImplicit = createAction('CREATE_ACCOUNT_FROM_IMPLICIT', async (accountId, implicitAccountId, recoveryMethod) => {
    const recoveryKeyPair = await wallet.keyStore.getKey(wallet.connection.networkId, implicitAccountId)
    if (recoveryKeyPair) {
        await wallet.saveAccount(accountId, recoveryKeyPair)
    }
    const publicKey = new PublicKey({ keyType: KeyType.ED25519, data: Buffer.from(implicitAccountId, 'hex') })
    await wallet.createNewAccount(accountId, { fundingAccountId: implicitAccountId }, recoveryMethod, publicKey)
})
```

instead it makes sense to create another action similar to this one which would use your current account instead of `implicitAccountId`

downstream in `wallet` there is already handling for implicit vs non-implicit:

```
    async createNewAccountFromAnother(accountId, fundingAccountId, publicKey) {
        const account = await this.getAccount(fundingAccountId)
        await account.functionCall(ACCOUNT_ID_SUFFIX, 'create_account', {
            new_account_id: accountId,
            new_public_key: publicKey.toString().replace(/^ed25519:/, '')
            // TODO: Adjust gas if necessary
            // TODO: Should new account have other than minimum balance? For implicit account?
        }, LINKDROP_GAS, MIN_BALANCE_FOR_GAS)
        if (!this.accounts[fundingAccountId]) {
            // Temporary implicit account used for funding – move whole balance by deleting it
            await account.deleteAccount(accountId)
        }
    }
```

Reply:
Good point I will update!